src/main/java/com/thebois/models/world/structures/House.java

```diff
@@ -31,4 +31,9 @@ public House deepClone() {
         return new House(getPosition());
     }
 
+    @Override
+    public float getCost() {
+        return Float.MAX_VALUE;
+    }
```

Comment:
Same as terrain, wouldn't it be better to move this up to `Abstract Structre` since this should be the same for most structures.

Reply:
Either way we go, it should be the same for all `ITile` implementations.